rali/rali/source/fused_crop_decoder.cpp

```diff
@@ -0,0 +1,178 @@
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <stdio.h>
+#include <commons.h>
+#include <string.h>
+#include "fused_crop_decoder.h"
+
+FusedCropTJDecoder::FusedCropTJDecoder(){
+    m_jpegDecompressor = tjInitDecompress();
+
+#if 0
+    int num_avail_scalings = 0;
+    auto scaling_factors = tjGetScalingFactors	(&num_avail_scalings);	
+    for(int i = 0; i < num_avail_scalings; i++) {
+        if(scaling_factors[i].num < scaling_factors[i].denom) {
+
+            printf("%d / %d  - ",scaling_factors[i].num, scaling_factors[i].denom );
+        }
+    }
+#endif
+};
+
+Decoder::Status FusedCropTJDecoder::decode_info(unsigned char* input_buffer, size_t input_size, int* width, int* height, int* color_comps) 
+{
+    //TODO : Use the most recent TurboJpeg API tjDecompressHeader3 which returns the color components
+    if(tjDecompressHeader2(m_jpegDecompressor,
+                            input_buffer, 
+                            input_size, 
+                            width, 
+                            height, 
+                            color_comps) != 0)
+    {
+        WRN("Jpeg header decode failed " + STR(tjGetErrorStr2(m_jpegDecompressor)))
+        return Status::HEADER_DECODE_FAILED;
+    }
+    return Status::OK;
+}
+
+Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t input_size, unsigned char *output_buffer,
+                                  size_t max_decoded_width, size_t max_decoded_height,
+                                  size_t original_image_width, size_t original_image_height,
+                                  size_t &actual_decoded_width, size_t &actual_decoded_height,
+                                  Decoder::ColorFormat desired_decoded_color_format, DecoderConfig decoder_config, bool keep_original_size)
+{
+    int tjpf = TJPF_RGB;
+    int planes = 1;
+    switch (desired_decoded_color_format) {
+        case Decoder::ColorFormat::GRAY:
+            tjpf = TJPF_GRAY;
+            planes= 1;
+        break;
+        case Decoder::ColorFormat::RGB:
+            tjpf = TJPF_RGB;
+            planes = 3;
+        break;
+        case Decoder::ColorFormat::BGR:
+            tjpf = TJPF_BGR;
+            planes = 3;
+        break;
+    };
+    actual_decoded_width = max_decoded_width;
+    actual_decoded_height = max_decoded_height;
+    
+    std::vector<float> crop_mul_param =  decoder_config.get_crop_param();
+    auto is_valid_crop = [](uint h, uint w, uint height, uint width)
+    {
+        return (h < height && w < width); 
+    };
+    int num_of_attempts = 5;
+    double target_area;
```

Comment:
Codacy found an issue: [The scope of the variable 'target_area' can be reduced.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5763058)